compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -456,6 +459,24 @@ private DiagnosticPos getPosition(Node node) {
                 endPos.offset() + 1);
     }
 
+    private DiagnosticPos getPositionWithoutMetadata(Node node) {
+        if (node == null) {
+            return null;
+        }
+        NonTerminalNode nonTerminalNode = (NonTerminalNode) node;
+        ChildNodeList children = nonTerminalNode.children();
+        // If there's metadata it will be the first child.
+        // Hence set start position from next immediate child.
+        if (children.get(0).kind() == SyntaxKind.METADATA) {
+            LinePosition startPos = children.get(1).lineRange().startLine();
+            LinePosition endPos = node.lineRange().endLine();
+            return new DiagnosticPos(diagnosticSource, startPos.line() + 1, endPos.line() + 1, startPos.offset() + 1,
+                    endPos.offset() + 1);
+        } else {
+            throw new IllegalArgumentException("Node should have metadata");
```

Comment:
Shall we return the normal-positions if there are no metadata?
Reason is, theres a plan for making metadata optional in the syntax tree.